src/TinyCiv.Client/MainWindow.xaml.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Controls.Primitives;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using TinyCiv.Client.Code;
+using TinyCiv.Client.Code.units;
+using TinyCiv.Server.Client;
+using TinyCiv.Shared.Events.Client;
+using TinyCiv.Shared.Events.Server;
+using TinyCiv.Shared.Game;
+
+namespace TinyCiv.Client
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        IServerClient Client;
+        private GameGrid gameGrid;
+        private Player currentPlayer;
+
+        private bool isUnitSelected = false;
+        private int unitIndex;
+
+        public MainWindow()
+        {
+            InitializeComponent();
+
+            Loaded += PlayerConnection;
+            Closed += PlayerDisconnect;
+        }
+        private async void PlayerConnection(object sender, RoutedEventArgs e)
+        {
+            await Task.Run(async () =>
+            {
+                Client = ServerClient.Create("http://localhost:5000");
+
+                // Subscribe to events after creating the client
+                Client.ListenForNewPlayerCreation(OnPlayerJoin);
+                Client.ListenForGameStart(OnGameStart);
+                Client.ListenForMapChange(OnMapChange);
+
+                await Client.SendAsync(new JoinLobbyClientEvent());
+            });
```

Comment:
Creating a long running thread from Task.Run is bad practice, since you make one of the threads from the thread pool always busy even thought their purpose is to "quickly" do background jobs and go back to "waiting" state

Can keep as is or create a thread from ```Thread``` class

Reply:
Also nice catch, I would not have guessed that WPF single thread would block initial connection to SignalR, thought at least initial "handshake" would work